inferno-ml/src/Inferno/ML/Types/Value.hs

```diff
@@ -9,30 +9,33 @@ import Language.Haskell.TH.Quote (QuasiQuoter (..))
 import Prettyprinter (Pretty (pretty), align)
 import qualified Torch as T
 
-data MlValue
+data MlValue x
   = VTensor T.Tensor
   | VModel T.ScriptModule
+  | VExtended x
```

Comment:
This is pretty awful and hacky, but I can't think of a better way to allow additional types (which is necessary in my WIP `inferno-ml-server` implementation). The problem is that `VCustom` is set to `MlValue`, but we need the custom type to be able to represent additional things (not just `MlValue`s). I guess it would be possible to create another type that can hold an `MlValue` or another type, but then it would be necessary to map over all of the `ML` prelude functions. Or perhaps it might be possible once I add the more polymorphic `ML` prelude

If anyone has a better idea, please let me know!

Reply:
Yeah I knew this would be a problem when I introduced `VCustom`... Is there a way to use a typeclass to make this nicer? E.g.
```haskell
zerosFun :: (MonadThrow m, CustomValue c, HasTensorValue c) => Value c m
```

Reply:
Yes, that could work. Maybe something like

```haskell
class HasMlTypes c where
  type Tensor c
  type Model c
  ... 
```

would work for both of the required types